### PR TITLE
#14406: CCL Perf Monitoring

### PR DESCRIPTION
**Pull request by Aswinmcw**
### Ticket
#14406 

### Problem description
CCL ops needs to get profiled

### What's changed
Script added to get the perf report

- Total iterations = 20
- Average computation except first 4 iterations

Screenshot 2024-10-30 at 12 09 09 PM

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by SeanNijjar** [comment]
This looks great! Can you please provide a summary table that collapse the measured samples to a single entry per test so that when we have many test cases it's more consumable,readable?

File: tests/ttnn/unit_tests/operations/ccl/perf/test_all_gather_N300_post_commit.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    "num_devices, num_links, output_shape, dim, layout",
+    [
+        (2, 1, [1, 1, 64, 16384], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [8, 5, 32, 768], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [1, 1, 32, 736], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [1, 1, 32, 704], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [1, 1, 64, 704], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [1, 1, 32, 736], 3, ttnn.ROW_MAJOR_LAYOUT),
+        (2, 1, [1, 1, 32, 704], 3, ttnn.ROW_MAJOR_LAYOUT),
+        # (2, 1, [1, 1, 64, 704], 3, ttnn.ROW_MAJOR_LAYOUT),
```
Comment (SeanNijjar) [inline tests/ttnn/unit_tests/operations/ccl/perf/test_all_gather_N300_post_commit.py:25]: Can we remove these and other commented out cases? We can cleanup/choose specific cases after.

File: tests/ttnn/unit_tests/operations/ccl/perf/perf_run.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        "HOST DURATION [ns]": 10000,
+        "Cycles Count": 120000,
+        "OP TO OP LATENCY [ns]": 3500,
+        "DEVICE FW DURATION [ns]": 115000,
+        "DEVICE KERNEL DURATION [ns]": 110000,
+    }
+
+    averages = df.iloc[-1]
+
+    # for column, reference in reference_values.items():
```
Comment (SeanNijjar) [inline tests/ttnn/unit_tests/operations/ccl/perf/perf_run.py:22]: can this be deleted or should it be kept in?

**Review by SeanNijjar — APPROVED** [review]

**Comment by SeanNijjar** [comment]
Future changes:
- Add the equivalent to reduce scatter
- Add t3k and TG test functions.
- Add the computed BW columns (based on equations from master issue)

File: tests/ttnn/unit_tests/operations/ccl/perf/perf_run.py (hunk 2 above)
Comment (Aswinmcw) [reply tests/ttnn/unit_tests/operations/ccl/perf/perf_run.py:22]: Yes this can be removed

File: tests/ttnn/unit_tests/operations/ccl/perf/perf_csv.py
Comment (jvegaTT) [inline tests/ttnn/unit_tests/operations/ccl/perf/perf_csv.py:1]: Copyright on new files should be 2024

File: tests/ttnn/unit_tests/operations/ccl/perf/perf_run.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
```
Comment (jvegaTT) [inline tests/ttnn/unit_tests/operations/ccl/perf/perf_run.py:1]: Copyright on new files should be 2024

**Review by jvegaTT — APPROVED** [review]
Looks good

**Pull request merged**